File: gaming_forces/gaming.cpp
#include <iostream>
#include <sstream>

using namespace std;

int main()
{
    int case_num;
    cin >> case_num;
    for (int i = 0; i < case_num; i++)
    {
        int size;
        cin >> size;
        cin.ignore();
        string line;
        getline(cin, line);
        istringstream iss(line);
        int digit;
        int num_one_spells = 0;
        int num_ones = 0;
        int num_spells;
        int incremented = 0;
        while(iss >> digit)
        {
            if (digit == 1)
            {
                num_ones++;
                incremented = 1;
            }

            if (num_ones % 2 == 0 && incremented == 1)
            {
                num_one_spells++;
                incremented = 0;
            }
        }


        num_spells = size - (num_one_spells);

        cout << num_spells << "\n";
    }

}

File: gaming_forces/gaming.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        size = int(input())
        digits = input().split()
        digits = [int(i) for i in digits]
        num_ones = 0
        num_one_spells = 0
        count = 0
        for digit in digits:
            if digit == 1:
                num_ones += 1
                count = 1

            if num_ones % 2 == 0 and count == 1:
                num_one_spells += 1
                count = 0
                
        num_spells = size - num_one_spells
        print(num_spells)
    
    
    
    
    
    
    
    
    
    
    
    
    
if __name__ == "__main__":
    main()